py/server/tests/test_updateby.py

```diff
@@ -25,19 +27,39 @@ def tearDown(self) -> None:
         self.ticking_table = None
         super().tearDown()
 
-    def test_ema(self):
-        op_ctrl = OperationControl(on_null=BadDataBehavior.THROW,
-                                   on_nan=BadDataBehavior.RESET,
-                                   big_value_context=MathContext.UNLIMITED)
-
-        ops = [ema_tick_decay(time_scale_ticks=100, cols="ema_a = a"),
-               ema_tick_decay(time_scale_ticks=100, cols="ema_a = a", op_control=op_ctrl),
-               ema_time_decay(ts_col="Timestamp", time_scale=10, cols="ema_a = a"),
-               ema_time_decay(ts_col="Timestamp", time_scale="00:00:00.001", cols="ema_c = c",
-                              op_control=op_ctrl),
-               ]
+    em_op_ctrl = OperationControl(on_null=BadDataBehavior.THROW,
+                               on_nan=BadDataBehavior.RESET,
+                               big_value_context=MathContext.UNLIMITED)
+
+    em_ops = [
+        # exponential moving average
+        ema_tick_decay(time_scale_ticks=100, cols="ema_a = a"),
+        ema_tick_decay(time_scale_ticks=100, cols="ema_a = a", op_control=em_op_ctrl),
+        ema_time_decay(ts_col="Timestamp", time_scale=10, cols="ema_a = a"),
+        ema_time_decay(ts_col="Timestamp", time_scale="00:00:00.001", cols="ema_c = c",
+                       op_control=em_op_ctrl),
+        # exponential moving sum
+        ems_tick_decay(time_scale_ticks=100, cols="ema_a = a"),
+        ems_tick_decay(time_scale_ticks=100, cols="ema_a = a", op_control=em_op_ctrl),
+        ems_time_decay(ts_col="Timestamp", time_scale=10, cols="ema_a = a"),
+        ems_time_decay(ts_col="Timestamp", time_scale="00:00:00.001", cols="ema_c = c",
+                       op_control=em_op_ctrl),
+        # exponential moving minimum
+        emmin_tick_decay(time_scale_ticks=100, cols="ema_a = a"),
+        emmin_tick_decay(time_scale_ticks=100, cols="ema_a = a", op_control=em_op_ctrl),
+        emmin_time_decay(ts_col="Timestamp", time_scale=10, cols="ema_a = a"),
+        emmin_time_decay(ts_col="Timestamp", time_scale="00:00:00.001", cols="ema_c = c",
+                       op_control=em_op_ctrl),
+        # exponential moving maximum
+        emmax_tick_decay(time_scale_ticks=100, cols="ema_a = a"),
+        emmax_tick_decay(time_scale_ticks=100, cols="ema_a = a", op_control=em_op_ctrl),
+        emmax_time_decay(ts_col="Timestamp", time_scale=10, cols="ema_a = a"),
+        emmax_time_decay(ts_col="Timestamp", time_scale="00:00:00.001", cols="ema_c = c",
+                       op_control=em_op_ctrl),
+    ]
```

Comment:
It reads a bit weird when the class variables are scattered around in the code. The benefit of doing so is obvious in that it has the variables defined right before they are used in the testing methods. From a code organization POV it might be better to fold them all into a setupClass() method (see test_arrow.py for an example). The current per-test fixtures (in setUp()/tearDown()) could also be moved there since they don't need to be refreshed for each test case.
This is just a suggestion as it is not so clear cut better.

Reply:
I like the suggestion, have refactored to move those lists to a common section.